tests/IVIMmodels/unit_tests/test_ivim_fit.py

```diff
@@ -161,14 +62,35 @@ def data_ivim_fit_saved():
 
 @pytest.mark.parametrize("name, bvals, data, algorithm, xfail, kwargs, tolerances", data_ivim_fit_saved())
 def test_ivim_fit_saved(name, bvals, data, algorithm, xfail, kwargs, tolerances, request):
+    global test_results
     if xfail["xfail"]:
         mark = pytest.mark.xfail(reason="xfail", strict=xfail["strict"])
         request.node.add_marker(mark)
     fit = OsipiBase(algorithm=algorithm, **kwargs)
     signal, ratio = signal_helper(data["data"])
+    
     tolerances = tolerances_helper(tolerances, ratio, data["noise"])
     [f_fit, Dp_fit, D_fit] = fit.osipi_fit(signal, bvals)
+    def to_list_if_needed(value):
+        return value.tolist() if isinstance(value, np.ndarray) else value
+    test_result = {
+        "name": name,
+        "algorithm": algorithm,
+        "f_fit": to_list_if_needed(f_fit),
+        "Dp_fit": to_list_if_needed(Dp_fit),
+        "D_fit": to_list_if_needed(D_fit),
+        "f": to_list_if_needed(data['f']),
+        "Dp": to_list_if_needed(data['Dp']),
+        "D": to_list_if_needed(data['D']),
+        "status": "PASSED"
+    }
+    if xfail["xfail"]:
+        test_result['status'] = "XFAILED"
```

Comment:
What happens if this test fails? I'm not sure actually if it would continue to run or stop, but the `test_result` clearly only has `PASS` and `XFAILED`.

Reply:
For the current setup it fails on the CI, is that correct? Or should it be marked on the file as `Failed` and the test should pass?
I tried changing the tolerance values and it fails for some cases with that error for example:
```
E           AssertionError: 
E           Not equal to tolerance rtol=0.05, atol=0.01
E           
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 0.03707215
E           Max relative difference: 0.3462352
E            x: array(0.07)
E            y: array(0.107072)
```
<img width="808" alt="image" src="https://github.com/user-attachments/assets/dd7b2142-7df3-47c5-adba-ab9af33069f3">

Reply:
I don't think the test should be failing currently. If you changed it to fail then that makes sense.

But if it fails does it write the file? And if it does write the file it doesn't correctly report the failed test because the only two options for `test_result['status']` are `PASSED` and `XFAILED`. This is why I'm strongly suggesting using a package or some method that is designed to do this rather than trying to force something.

I'm also noticing that you're not doing anything with this value. Ideally the plots would have colored points based on the status, for example green for pass, blue for xfail, red for fail.